# scMultiome
## 从单细胞转录组走向多组学研究
![](image/outline.png)
## 3端单细胞全长免疫组库测序
![](image/vdj.png)
## 单细胞TCR测序研究PD-1治疗前后T细胞克隆型变化
![](image/tcr.png)
## 单细胞BCR测序寻找新冠肺炎中和抗体
![](image/bcr.png)
## 单细胞靶向捕获突破3‘端的限制
![](image/3'_limitation.png)
单细胞靶向捕获：定制化的引物序列（替代polyA尾巴）进而特性行的捕获到想要研究的基因的基因突变和融合信息。
## 应用示例：肿瘤细胞肺癌驱动基因状态
![](image/example.png)
肺癌相关的驱动基因的上下游引物设计成一个panel，通过这个panel可以检测基因突变的状态和mRNA全转录组的信息
## 单细胞转录组综合指标分数可预测治疗疗效
![](image/application.png)
## 结合药物信息解读测序结果，辅助指定用药方案
![](image/application1.png)
## 单细胞动态转录组技术增加时间纬度
![](image/time-d.png)

```Cell Ranger ARC``` is a set of analysis pipelines that process Chromium Single Cell Multiome ATAC + Gene Expression sequencing data to generate a variety of analyses pertaining to gene expression, chromatin accessibility and their linkage. Furthermore, since the ATAC and gene expression measurements are on the very same cell, we are able to perform analyses that link chromatin accessibility and gene expression.

```cellranger-arc mkfastq``` demultiplexes raw base call (BCL) files generated by Illumina sequencers into FASTQ files. It is a wrapper around Illumina's bcl2fastq, with additional useful features that are specific to 10x libraries and a simplified sample sheet format. The same command can be used to demultiplex both ATAC and GEX flow cells.

```cellranger-arc count``` takes FASTQ files from cellranger-arc mkfastq and performs alignment, filtering, barcode counting, peak calling and counting of both ATAC and GEX molecules. Furthermore, it uses the Chromium cellular barcodes to generate feature-barcode matrices, perform dimensionality reduction, determine clusters, perform differential analysis on clusters and identify linkages between peaks and genes. The count pipeline can take input from multiple sequencing runs on the same GEM well.

```cellranger-arc aggr``` aggregates and analyzes the outputs from multiple runs of cellranger-arc count (such as from multiple samples from one experiment). Features include normalization of input runs to same median fragments per cell (sensitivity), detection of accessible chromatin peaks, count matrix generation for peaks and transcription factors for the aggregate data, dimensionality reduction, cell clustering, and cluster differential accessibility analysis.

```cellranger-arc``` reanalyze takes the analysis files produced by cellranger-arc count or cellranger-arc aggr and reruns secondary analysis. Features include tunable parameter settings related to cell calling, dimensionality reduction, cell clustering, and cluster differential accessibility analysis.

These pipelines combine Chromium-specific algorithms with the widely used aligners STAR and BWA. Output is delivered in standard BAM, MEX, CSV, HDF5 and HTML formats that are augmented with cellular information and a .cloupe file for use with Loupe Browser.


**Workflows**

If you are beginning with raw base call (BCL) files, the Cell Ranger ARC workflow starts with demultiplexing the BCL files for each flow cell directory for all relevant ATAC and GEX sequencing runs. 10x recommends using cellranger-arc mkfastq as described in Generating FASTQs. If you are beginning with FASTQ files that have already been demultiplexed with bcl2fastq directly, or from a public source such as SRA, you can skip cellranger-arc mkfastq and begin with cellranger-arc count. Please see the Specifying Input FASTQs page for specific guidelines on which arguments to use for your scenario.

The exact steps of the workflow vary depending on how many samples, GEM wells, and flow cells you have. This section describes the simplest possible workflows.

**One Sample, One GEM Well, One ATAC + One GEX Flow cell**

![](image/arc-compute-workflow-one-flowcell.png)

In this example you have one sample that is processed through one GEM well (a set of partitioned cells from a single 10x Chromium™ Chip channel) and results in one Multiome ATAC library and one Multiome GEX library. Each library is sequenced separately on one flow cell. In this case you would generate FASTQs separately for ATAC and GEX by running cellranger-arc mkfastq on the respective flow cells and run cellranger-arc count as described in Single-Sample Analysis.

**One Sample, One GEM well, Multiple ATAC + Multiple GEX Flow cells**

![](image/arc-compute-workflow-two-flowcells.png)

In this example you have one sample that is processed through one GEM well resulting in one ATAC library and one GEX library. The ATAC and GEX libraries are sequenced on two flow cells each. As an example, this may be done to increase sequencing depth, when the first sequencing run did not produce enough raw read pairs per cell. Here we would run cellranger-arc mkfastq a total of four times: once for each of the two ATAC flow cells and once for each of the two GEX flow cells. All of the reads can be combined in a single instance of the cellranger-arc count pipeline. This process is described in Specifying Input Fastqs.